metaflow/linters/linter_interface.py

```diff
@@ -0,0 +1,51 @@
+from abc import ABC
+from enum import Enum
+from typing import Union
+
+from metaflow.metaflow_environment import MetaflowEnvironment
+from metaflow.plugins.pypi.conda_environment import CondaEnvironment
+
+
+class LinterConfig:
```

Comment:
@savingoyal just FWIW should we support 2 or more, or perhaps replace `ruff` with a better technique in time to come, a not-so-factory and factory instantiated pattern may get us better code quality.